### ISSUE TITLE
[Question] Product Variants

### ISSUE THREAD

**Issue by brianjhanson** [verbb/gift-voucher#34]
We're updating a shop from Craft 2 to 3 and evaluating moving our custom gift card implementation over to Gift Voucher. In our current shop, we have a single "Gift Card" product with several variants for the different price levels ($25, $50, $75, etc). 

It looks like Gift Vouncher doesn't currently support variants, so to move over, we would have to create create a Voucher for each price level. Is that correct? Since the content for each price level is the same, it would be great to have a single gift card product with a variant for each price, but I'm not seeing how best to achieve this with Gift Voucher. Is that possible at all? 

Thank you!

**Comment by engram-design** [comment]
Hey @brianjhanson that's pretty much it. You'll need to create a voucher for each set price you like. As such, you're really creating multiple voucher "products" with a single "variant". That's not what's going on, but conceptually in the context of Commerce, you can think of it like that.

Otherwise, you could also setup a single voucher for variable price, where the customer can define the price. That just comes down to how your site and client wants it to function. It might be better to have set vouchers to pick from.

So at the moment, its not possible, and unfortunately, yes - it does mean duplicate content.

**Comment by brianjhanson** [comment]
@engram-design thanks for the response! I ended up moving our old Gift Card from a product to a single and just adding a Gift Voucher field to it. That way the single controls the content, but the user purchases one of the vouchers. 

I would love variants in the future (obviously) but at the moment this works pretty well. Thanks!

**Comment by engram-design** [comment]
Nice solution! It's probably not something on the immediate list, as I know the Commerce crew are looking to include Gift Voucher functionality natively.

**Comment by pixelmachine** [comment]
@engram-design as far as I can see there doesn't seem to have been much movement or even discussion about native vouchers in Commerce since this issue was created. 

If that is the case and it's not just something I've missed would you reconsider the idea of adding variants to Gift Voucher? Would be very useful and would massively simplify templating.

**Comment by pixelmachine** [comment]
@engram-design been a few years now since this was closed and I'm still regularly wishing that vouchers were equivalent to products. There are a lot of cases where significant additional work needs to happen to accommodate a voucher because of the lack of defaultVariant and other such structural things.

There are still no concrete plans as far as I can find to integrate vouchers into Craft Commerce. Nothing on the roadmap and no mention on Discord or Github (except a single discussion that's had no input). If gift vouchers are coming I assume it will be years off still.

Unless you know different?

**Comment by engram-design** [comment]
Indeed, the last I heard about this was years ago, and it was pretty high on their list to implement and they deemed it a core feature. Likely that's changed (it seems)!

Happy to re-open this one in that regard, but still no ETA on implementing this one. We're prioritizing Craft 5/Commerce 5 support first, then we'll see how we go from there. I'll admit I've not looked into it extensively, but this would likely be significant work and a breaking change to existing installs.
